homeassistant/components/backup/util.py

```diff
@@ -122,7 +122,7 @@ def read_backup(backup_path: Path) -> AgentBackup:
 def suggested_filename_from_name_date(name: str, date_str: str) -> str:
     """Suggest a filename for the backup."""
     date = dt_util.parse_datetime(date_str, raise_on_error=True)
-    return "_".join(f"{name} - {date.strftime('%Y-%m-%d %H.%M %S%f')}.tar".split())
+    return "_".join(f"{name}-{date.strftime('%Y-%m-%d %H-%M-%S')}.tar".split())
```

Comment:
I think underscore should be used to separate each thing. Ie use underscore to separate the end of the name from the date.

Reply:
you mean like this `Automatic_backup_2025.2.0.dev0_2025-01-09_20-14-35` ?

Reply:
further, what do you think about the order of the parts? 
1. name,version,date: `Automatic_backup_2025.2.0.dev0_2025-01-09_20-14-35`
2. date,name,version: `2025-01-09_20-14-35_Automatic_backup_2025.2.0.dev0`
3. version,date,name: `2025.2.0.dev0_2025-01-09_20-14-35_Automatic_backup`
4. version,name,date: `2025.2.0.dev0_Automatic_backup_2025-01-09_20-14-35`

Reply:
The version is part of the name, I think. I don't have a strong opinion. It could be nice to sort the files by date by default, even if another name template is used than the one for automatic backups. :shrug:

Reply:
I think we should stick with 1 for now. We have ideas to see if we want to allow customization in the future, but that is not for now.

The reason to stick with 1 is that it actually covers more use cases right now. The current naming allows sorting by name to group/sort similar types of backups, while the file's metadata can just be sorted by the file system (e.g., sorting on the file's creation date).

Reply:
ok ... just to be sure - removing the milliseconds is OK, or were they added on purpose?

Reply:
My preference would be 1 as well. But I'd prefer to not have underscores. All file systems allow spaces. Having all these underscores makes it hard to read. So my preference would be: `Automatic backup 2025.2.0.dev0 2025-01-09 20-14-35.tar`

For Google Drive, this format can change anytime without being a breaking change. Because the Google Drive integration has platinum quality scale, I'm approving this now so that you can merge it if needed without needing my approval or an admin to merge. But do wait for someone else to approve before merging, and after you replace '-' with ' ' between name and date.